### ISSUE TITLE
Enhancements for the Upgrade Command

### ISSUE THREAD

**Issue by tareknaser360** [coffee-tools/coffee#187]
Currently, the `upgrade` command provides a way to upgrade an entire repository. However, there are several improvements that can be made to enhance the functionality and flexibility of the command. This is a tracking issue of  the proposed enhancements:
[ ] Upgrade a Single Plugin: `coffee upgrade <plugin_name>`
[ ] Upgrade All Plugins: `coffee upgrade --all`
[ ] Verbose Flag. `coffee upgrade <plugin_name> --verbose`
[ ] Specify Branch for Upgrading a repository cloned version: `coffee upgrade <repo_name> --branch <branch_name>`

**Comment by tareknaser360** [comment]
I am now thinking that `coffee upgrade <repo_name> --branch <branch_name>` will be a bit confusing for the user because all other upgrade functions expect the user to pass the name of the plugin not the name of the repository.

But it also doesn't make sense to have `coffee upgrade <plugin_name> --branch <branch_name>` because the branch is specific to the repository not the plugin. One repository can and probably has multiple plugins.

**Comment by vincenzopalazzo** [comment]
This is a good point we should think about it

**Comment by vincenzopalazzo** [comment]
Ok there are two case here;

- Upgrade all the repository `coffee upgrade <repo_name> --branch <branch_name>`
- Upgrade only a plugin `coffee upgrade --plugin <plugin_name> --branch <branch_name>`: this is the complex case and it is better avoid this because we do not know if the logic works.

We should start with a simple solution and then iterate over it

**Comment by tareknaser** [comment]
> We should start with a simple solution and then iterate over it

agreed. I will start with `coffee upgrade <repo_name> --branch <branch_name>` 
Could you take a look at the updated issue description and confirm that this is what we want on the long term?

**Comment by vincenzopalazzo** [comment]
YEah looks good, the last points looks a little bit strange to me, but I think when we start to solve the problem it will be more clear, so all good for now

**Comment by tareknaser** [comment]
I think it makes sense for `coffee upgrade` to only work on repositories, not plugins. Adding a command like `coffee upgrade --plugin <plugin_name>` would complicate the code because it would create discrepancies between the remote git HEAD and the plugin's git HEAD.

it's best to stick with supporting repositories containing only one plugin.
cc @vincenzopalazzo

**Comment by vincenzopalazzo** [comment]
>I think it makes sense for coffee upgrade to only work on repositories, not plugins. 

A repository can be a plugin, e.g: folgore

**Comment by vincenzopalazzo** [comment]
idk I am still confusing about the upgrade command I should think more about it, maybe while using it

**Comment by daywalker90** [comment]
i think `upgrade` currently does not re-parse the manifest file. I think it should!
